tests/components/aws/test_init.py

```diff
@@ -266,3 +266,51 @@ async def test_credential_skip_validate(hass):
     session = sessions.get("key")
     assert isinstance(session, MockAioSession)
     session.get_user.assert_not_awaited()
+
+
+async def test_service_call_extra_data(hass):
+    """Test service call extra data are parsed properly."""
+    with async_patch("homeassistant.components.aws.AioSession", new=MockAioSession):
+        await async_setup_component(
+            hass,
+            "aws",
+            {
+                "aws": {
+                    "notify": [
+                        {
+                            "service": "sns",
+                            "name": "SNS Test",
+                            "region_name": "us-east-1",
+                        }
+                    ]
+                }
+            },
+        )
+        await hass.async_block_till_done()
+
+    sessions = hass.data[aws.DATA_SESSIONS]
```

Comment:
Patch the library instead, to be able to assert mock calls. We don't want to access details like hass.data in the tests.

Reply:
This pattern is used in all the tests, we can fix that in a separate PR I think.

Reply:
https://github.com/home-assistant/core/pull/64943